ports/nrf/main.c

```diff
@@ -251,6 +256,11 @@ int main(int argc, char **argv) {
         }
     }
 
+    #if BLUETOOTH_SD
+    printf("MPY: soft reboot\n");
+    goto ble_reset;
```

Comment:
So this is no longer doing a real soft-reset, which would ordinarily mean that the uPy heap is reset.  I really think a proper soft-reset is important, otherwise it's not useful.

Reply:
Yes, that is correct. On the other hand, the patch is wrong, it was suppose to be "#if MICROPY_PY_BLE_NUS". But it gave the same result, as i was running over NUS in any case. Sorry about that.

The problem I'm trying to solve is how to get pyboard.py/ampy/rshell and friends work over a UART like devices on the PC. So that you can modify files etc. over the Bluetooth LE link. The problem is how `socat` times out the session due to inactivity when BLE reestablishes the link. This is why i shortened it to stay in the link and still give the PC tools the needed "MPY: soft reboot" message.

I'll drop this commit, take a new look at it, to see if i get something working that does full BLE re-connect.

Reply:
Ok.  In the meantime this PR now looks good to merge.  Shall I merge it?